libsolidity/formal/SMTEncoder.cpp

```diff
@@ -407,19 +407,25 @@ void SMTEncoder::endVisit(TupleExpression const& _tuple)
 		auto const& symbTuple = dynamic_pointer_cast<smt::SymbolicTupleVariable>(m_context.expression(_tuple));
 		solAssert(symbTuple, "");
 		auto const& symbComponents = symbTuple->components();
-		auto const& tupleComponents = _tuple.components();
-		solAssert(symbComponents.size() == _tuple.components().size(), "");
+		auto const* tupleComponents = &_tuple.components();
+		while (tupleComponents->size() == 1)
+		{
+			auto singleTuple = dynamic_pointer_cast<TupleExpression>(tupleComponents->front());
```

Comment:
What if the single tuple element is not itself a tuple? Should this rather e.g. be
```suggestion
			if (auto const* singleTuple = dynamic_cast<TupleExpression const*>(tupleComponents->front().get()))
				tupleComponents = &singleTuple->components();
			else
				break;
```
or something like that?

Do you have a test case like
```
pragma experimental SMTChecker;

contract C {
	function f() public pure {
		(2);
	}
}
```
and won't that throw in the next line, since ``singleTuple`` will be ``nullptr``?

Reply:
In that case the type of the tuple is actually the type of the inner expression, so it still works. Added an assertion and the test.